lib/fathead/react_js/parse.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import os
+from bs4 import BeautifulSoup
+
+DOWNLOADED_HTML_PATH = 'download/'
+REACT_API_DOCS_URL = "https://facebook.github.io/react/docs/"
+
+
+class HtmlFileData(object):
+    """
+    Class to Hold Html Data fetched from React Docs
+    """
+    def __init__(self, file):
+        """
+        Initialize HtmlFileData object. Load data from Downloaded docs
+        """
+        self.HTML = ""
+        self.FILE = file
+        self.load_data()
+
+    def load_data(self):
+        """
+        Open HTML File and load data
+        """
+        with open(self.FILE, 'r') as html_file:
+            document = html_file.read()
+            self.HTML = document
+
+    def get_raw_data(self):
+        """
+        Returns the Plain HTML
+        """
+        return self.HTML
+
+    def get_file(self):
+        """
+        Return File Object
+        """
+        return self.file
+
+
+class APIDocsParser(object):
+    """
+    Parses the HTML Data and fetches
+    title , content, example code snippet
+    """
+    def __init__(self, data):
+        """
+        Initialize APIDocsParser object with API Reference HTML
+        """
+        self.data = self.get_api_reference_html(data)
+        self.parsed_data = []
+
+    def get_api_reference_html(self, data):
+        """
+        Parses the HTML File Text and returns the API Reference soup
+        """
+        soup = BeautifulSoup(data.get_raw_data(), 'html.parser')
+        reference = soup.find('a', attrs={'name': 'reference'})
+        if reference:
+            reference_soup = reference.find_parent().find_parent()
+            return reference_soup
+
+    def parse_data(self, file_data):
+        """
+        Parses Individual API and extracts Title , Link , 
+        Content and Example Code
+        """
+        if self.data:
+            all_api_reference = self.data.findAll(["h4","h3"])
+            for api in all_api_reference: 
+                title = api.text.replace(' #', '')
+
+                href = self.parse_link(file_data, api)
+
+                content = self.parse_content(api)
+
+                example = self.parse_example(api)
+
+                section = {
+                    'title': title,
+                    'href': href,
+                    'content': content,
+                    'example': example
+                }
+                self.parsed_data.append(section)
+
+    def parse_example(self, api):
+        """
+        Extract the Example Code Snippet
+        """
+        example_code = ''
+        for tag in api.next_siblings:
+            if tag.name == 'div':
+                code = str(tag.find('pre'))
+                if code != 'None':
+                    example_code = code
+        return example_code.strip('\n')
```

Comment:
Sorry for missing this. I will fix it
@MariagraziaAlastra  Are there any more mistakes ?